### ISSUE TITLE
RUF100 fix removes remaining comment

### ISSUE THREAD

**Issue by adamtheturtle** [astral-sh/ruff#20762]
### Summary

I use `ruff` and `pylint`.
I used to have to ignore errors for each on the same line, so I added the following ignore:

```python
"""Demonstrate ruff removal of pylint disable."""

import ast


class A(ast.NodeVisitor):
    """Subclass of ast.NodeVisitor."""

    def visit_ImportFrom(  # noqa: N802, pylint: disable=invalid-name
        self,
        node: ast.ImportFrom,
    ) -> None:
        """Override method."""
```

`ruff check --select=ALL ruff_example.py` gives:

```
RUF100 [*] Unused `noqa` directive (unused: `N802`)
  --> ruff_example.py:9:28
   |
 7 |     """Subclass of ast.NodeVisitor."""
 8 |
 9 |     def visit_ImportFrom(  # noqa: N802, pylint: disable=invalid-name
   |                            ^^^^^^^^^^^^
10 |         self,
11 |         node: ast.ImportFrom,
   |
help: Remove unused `noqa` directive

Found 1 error.
[*] 1 fixable with the `--fix` option.
```

Adding `--fix` unexpectedly removes the `pylint` disable part of the comment (meaning that `pylint` errors).

### Version

0.14.0

**Comment by ntBre** [comment]
I was looking through other `RUF100` issues, and it looks like this was actually an intentional change in https://github.com/astral-sh/ruff/issues/12251 to assume that the rest of the comment is a description of the noqa code. The recommended workaround is to use a second `#` in the comment to separate the suppression comments like:

```py
def visit_ImportFrom(  # noqa: N802 # pylint: disable=invalid-name
```

`RUF100` won't delete past the second `#` in this case: [playground](https://play.ruff.rs/363341bf-d6a1-42e3-b367-39ec140dab6d).

**Comment by adamtheturtle** [comment]
Thank you for taking a look.

I think that this should be documented in https://docs.astral.sh/ruff/rules/unused-noqa/.
I also think perhaps the fix should be marked unsafe in the case that it will remove other comments.

**Comment by ntBre** [comment]
Yeah, I think those both make sense!

**Comment by MichaReiser** [comment]
I agree with the documentation but I don't think the fix should be marked as unsafe. Marking fixes as unsafe comes at a cost, they stop working out of the box. In this case, the fix does work as intended and follows noqa semantics.
